wrappers/golang/core/slice.go

```diff
@@ -188,6 +194,14 @@ func (h HostSlice[T]) SizeOfElement() int {
 	return int(unsafe.Sizeof(h[0]))
 }
 
+func (h HostSlice[T]) AsPointer() unsafe.Pointer {
+	return unsafe.Pointer(&h[0])
+}
```

Comment:
As discussed, only return a pointer to the first element and let the function below be used to get the unsafe.Pointer around it